src/catalog/column_stats_catalog.cpp

```diff
@@ -96,74 +98,157 @@ bool ColumnStatsCatalog::InsertColumnStats(
+                           ColumnId::TABLE_ID);
+  expression::AbstractExpression *tb_oid_const_expr =
+      expression::ExpressionUtil::ConstantValueFactory(
+          type::ValueFactory::GetIntegerValue(table_id).Copy());
+  expression::AbstractExpression *tb_oid_equality_expr =
+      expression::ExpressionUtil::ComparisonFactory(
+          ExpressionType::COMPARE_EQUAL, tb_oid_expr, tb_oid_const_expr);
+
+  auto *col_id_expr =
+      new expression::TupleValueExpression(type::TypeId::INTEGER, 0,
+                                                    ColumnId::COLUMN_ID);
+  col_id_expr->SetBoundOid(catalog_table_->GetDatabaseOid(),
+                           catalog_table_->GetOid(),
+                           ColumnId::COLUMN_ID);
+  expression::AbstractExpression *col_id_const_expr =
+      expression::ExpressionUtil::ConstantValueFactory(
+          type::ValueFactory::GetIntegerValue(column_id).Copy());
+  expression::AbstractExpression *col_oid_equality_expr =
+      expression::ExpressionUtil::ComparisonFactory(
+          ExpressionType::COMPARE_EQUAL, col_id_expr, col_id_const_expr);
+
+  expression::AbstractExpression *db_and_tb =
+      expression::ExpressionUtil::ConjunctionFactory(
+          ExpressionType::CONJUNCTION_AND, db_oid_equality_expr,
+          tb_oid_equality_expr);
+  expression::AbstractExpression *predicate =
+      expression::ExpressionUtil::ConjunctionFactory(
+          ExpressionType::CONJUNCTION_AND, db_and_tb, col_oid_equality_expr);
+
+  std::vector<codegen::WrappedTuple> result_tuples =
+      GetResultWithCompiledSeqScan(column_ids, predicate, txn);
+
+  PELOTON_ASSERT(result_tuples.size() <= 1);  // unique
+  if (result_tuples.empty()) {
     return nullptr;
   }
 
+  codegen::WrappedTuple tuple = result_tuples[0];
+
   type::Value num_rows, cardinality, frac_null, most_common_vals,
       most_common_freqs, hist_bounds, column_name, has_index;
 
-  num_rows = tile->GetValue(0, ColumnStatsOffset::NUM_ROWS_OFF);
-  cardinality = tile->GetValue(0, ColumnStatsOffset::CARDINALITY_OFF);
-  frac_null = tile->GetValue(0, ColumnStatsOffset::FRAC_NULL_OFF);
-  most_common_vals = tile->GetValue(0, ColumnStatsOffset::COMMON_VALS_OFF);
-  most_common_freqs = tile->GetValue(0, ColumnStatsOffset::COMMON_FREQS_OFF);
-  hist_bounds = tile->GetValue(0, ColumnStatsOffset::HIST_BOUNDS_OFF);
-  column_name = tile->GetValue(0, ColumnStatsOffset::COLUMN_NAME_OFF);
-  has_index = tile->GetValue(0, ColumnStatsOffset::HAS_INDEX_OFF);
+  num_rows = tuple.GetValue(ColumnId::NUM_ROWS);
+  cardinality = tuple.GetValue(ColumnId::CARDINALITY);
+  frac_null = tuple.GetValue(ColumnId::FRAC_NULL);
+  most_common_vals = tuple.GetValue(ColumnId::MOST_COMMON_VALS);
+  most_common_freqs = tuple.GetValue(ColumnId::MOST_COMMON_FREQS);
+  hist_bounds = tuple.GetValue(ColumnId::HISTOGRAM_BOUNDS);
+  column_name = tuple.GetValue(ColumnId::COLUMN_NAME);
+  has_index = tuple.GetValue(ColumnId::HAS_INDEX);

```

Comment:
I remember Pooja mentioning in the last meeting that tuple.GetValue() is going to deprecated. However, I vaguely remember her mentioning that it could still be used with catalog tables. You might want to talk to her/Andy about this.